crates/storage/src/lazy/lazy_cell.rs

```diff
@@ -151,8 +151,22 @@ where
 
     fn clear_spread(&self, ptr: &mut KeyPtr) {
         let root_key = ExtKeyPtr::next_for::<Self>(ptr);
-        if let Some(entry) = self.entry() {
-            entry.clear_spread_root(root_key)
+        match <T as SpreadLayout>::REQUIRES_DEEP_CLEAN_UP {
+            true => {
+                // The inner cell needs to be cleared, no matter if it has
+                // been loaded or not. Otherwise there might be leftovers.
+                // Load from storage and then clear:
+                clear_spread_root_opt::<T, _>(root_key, || self.get())
+            }
+            false => {
+                // Clear without loading from storage:
+                let footprint = <T as SpreadLayout>::FOOTPRINT;
+                assert!(footprint <= 16); // magic number
```

Comment:
maybe we want to unify these magic numbers with some global constant. could be done in a follow up though.

Reply:
Good point, I'll do it in a follow-up.